src/ir/value.h

```diff
@@ -63,13 +73,19 @@ class BlockArgument : public NamedValue<Block> {
  public:
   using NamedValue<Block>::NamedValue;
   const Block& block() const { return element(); }
+  bool operator==(const BlockArgument& other) const {
+    return NamedValuesEq(*this, other);
+  }
 };
 
 // Indicates the result of an operation.
 class OperationResult : public NamedValue<Operation> {
  public:
   using NamedValue<Operation>::NamedValue;
   const Operation& operation() const { return element(); }
+  bool operator==(const OperationResult& other) const {
```

Comment:
Not a big deal, but you might be able to inherit the operator from `NamedValue<T>` with an `using` declaration.

Reply:
I like this suggestion, I'm doing it.

Reply:
Done.